Zend/zend_API.c

```diff
@@ -903,35 +905,34 @@ static int zend_parse_va_args(int num_args, const char *type_spec, va_list *va,
 		}
 	}
 
-	if (min_num_args < 0) {
+	/* with no optional arguments the minimum number of arguments must be the same as the maximum */
+	if (!have_optional_args) {
 		min_num_args = max_num_args;
 	}
 
 	if (have_varargs) {
 		/* calculate how many required args are at the end of the specifier list */
 		post_varargs = max_num_args - post_varargs;
-		max_num_args = -1;
```

Comment:
Maybe change this to `max_num_args = UINT32_MAX` and drop changes below?

Reply:
Done, this does indeed make the code nicer